src/datasets/arrow_dataset.py

```diff
@@ -1530,31 +1530,52 @@ def load_from_disk(
 
         fs_token_paths = fsspec.get_fs_token_paths(dataset_path, storage_options=storage_options)
         fs: fsspec.AbstractFileSystem = fs_token_paths[0]
-        # copies file from filesystem if it is remote filesystem to local filesystem and modifies dataset_path to temp directory containing local copies
-        dataset_dict_json_path = Path(dataset_path, config.DATASETDICT_JSON_FILENAME).as_posix()
-        dataset_info_path = Path(dataset_path, config.DATASET_INFO_FILENAME).as_posix()
-        if not fs.isfile(dataset_info_path) and fs.isfile(dataset_dict_json_path):
-            raise FileNotFoundError(
-                f"No such file or directory: '{dataset_info_path}'. Expected to load a Dataset object, but got a DatasetDict. Please use datasets.load_from_disk instead."
-            )
 
         if is_remote_filesystem(fs):
-            src_dataset_path = extract_path_from_uri(dataset_path)
-            dataset_path = Dataset._build_local_temp_path(src_dataset_path)
-            fs.download(src_dataset_path, dataset_path.as_posix(), recursive=True)
+            dest_dataset_path = extract_path_from_uri(dataset_path)
+            path_join = posixpath.join
+        else:
+            fs = fsspec.filesystem("file")
+            dest_dataset_path = dataset_path
+            path_join = os.path.join
+
+        dataset_dict_json_path = path_join(dest_dataset_path, config.DATASETDICT_JSON_FILENAME)
+        dataset_state_json_path = path_join(dest_dataset_path, config.DATASET_STATE_JSON_FILENAME)
+        dataset_info_path = path_join(dest_dataset_path, config.DATASET_INFO_FILENAME)
+        if fs.isfile(dataset_dict_json_path):
```

Comment:
`is_file` may do some http calls - maybe you can call it once for dataset_info_path and for dataset_state_json_path to save some time ?

Reply:
True, I didn't realize! I'll update that now 👍🏻